hq_superset/utils.py

```diff
@@ -182,13 +182,24 @@ def re_eval_roles(self, existing_roles, new_domain_role):
         ]
         return new_domain_roles + additional_roles
 
+    def _ensure_datasource_perm_created(self):
+        "This allows users to create and update virtual datasets"
+        permission = self.sm.find_permission_view_menu("can_save", "Datasource")
+        if not permission:
+            permission = self.sm.add_permission_view_menu("can_save", "Datasource")
+        return permission
+
     def sync_domain_role(self, domain):
         # This creates DB schema, role and schema permissions for the domain and
         #   assigns the role to the current_user
         self._ensure_schema_created(domain)
         permission = self._ensure_schema_perm_created(domain)
         role = self._ensure_domain_role_created(domain)
         self.sm.add_permission_role(role, permission)
+
+        ds_permission = self._ensure_datasource_perm_created()
+        self.sm.add_permission_role(role, ds_permission)
```

Comment:
Just checking if we should put this in the role created for the domain?
I would prefer if that role is only limited to allowing access to domain schema.

We could simply add a new role for this and add that to the user.

Though, i think its better to solve this via https://dimagi.atlassian.net/browse/SC-2853 so additional permissions are handled the same way. We could add this permission to the permissions listed [here](https://docs.google.com/document/d/1AybtXCthYwmq3tLl8ygD3tX1hD79wj9JKoTOtzpfoW4/edit#heading=h.jnpvevto1u08)

What do you think?

Reply:
Yes! I forgot that this was a thing. I agree, let's handle this through that feature

Reply:
thanks for the flexibility @SmittieC